src/types/request.ts

```diff
@@ -48,5 +48,5 @@ export type RequestConfig = {
    *
    * Defaults to: "proxyAddr.compile('loopback')"
    */
-  trustProxy: (address: string, distance: number) => boolean
+  trustProxy: ((address: string, distance: number) => boolean) | (() => boolean)
```

Comment:
Any reason we want to allow a function that does not accept any parameters? 

If its related to the following line

```
normalizedConfig.trustProxy = () => tpValue
```

Then we should fix the function vs allowing another function type.

```ts
normalizedConfig.trustProxy = (_, __) => tpValue
```